### PR TITLE
[2108][7721061] Broader group control for Repo, Environemt and SecureFile (Aanad's PR)

### PR DESCRIPTION
**Pull request by v-soukum**
## Description
</br>
 Brief description of the requested change.

## Checklist
- [ ] I have read the instructions mentioned in [Contribute to Code](/Contributing.md).
- [ ] I have read and understood the criteria described under [submitting changes](/Contributing.md#submitting-changes). 
- [ ] The title of the PR clearly describes the intent of the PR.
- [ ] This PR does not introduce any breaking changes to the code.

### CODE REVIEW

File: src/AzSK.ADO/Framework/Configurations/SVT/ADO/ADO.Project.json [hunk 1]
```diff
@@ -370,6 +370,59 @@
...
+      "Enabled": true
+    },
+    {
+      "ControlID": "ADO_Project_AuthZ_Restrict_Broader_Group_Access_on_Repo",
+      "Description": "Do not allow repositories to inherit excessive permissions for a broad group of users at project level.",
+      "Id": "Project400",
+      "ControlSeverity": "High",
+      "Automated": "Yes",
+      "MethodName": "CheckBroaderGroupInheritanceSettingsForRepo",
+      "Rationale": "If a broad group (e.g., Contributors) is configured with excessive permissions at a project level, they are inherited by individual variable groups in the project. Then the integrity of variable groups can be compromised by a malicious user",
```
Comment (arvindsingh-cloud) [inline src/AzSK.ADO/Framework/Configurations/SVT/ADO/ADO.Project.json:398]: Repositories

File: src/AzSK.ADO/Framework/Configurations/SVT/ADO/ADO.Project.json [hunk 2]
```diff
@@ -370,6 +370,59 @@
...
+      "Enabled": true
+    },
+    {
+      "ControlID": "ADO_Project_AuthZ_Restrict_Broader_Group_Access_on_SecureFile",
+      "Description": "Do not allow secure files to inherit excessive permissions for a broad group of users at project level.",
+      "Id": "Project410",
+      "ControlSeverity": "High",
+      "Automated": "Yes",
+      "MethodName": "CheckBroaderGroupInheritanceSettingsForSecureFile",
+      "Rationale": "If a broad group (e.g., Contributors) is configured with excessive permissions at a project level, they are inherited by individual variable groups in the project. Then the integrity of variable groups can be compromised by a malicious user",
```
Comment (arvindsingh-cloud) [inline src/AzSK.ADO/Framework/Configurations/SVT/ADO/ADO.Project.json:416]: check Rationale

File: src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.Project.ps1 [hunk 3]
```diff
@@ -2645,4 +2645,206 @@ class Project: ADOSVTBase
...
         return $controlResult
     }
+
+    hidden [ControlResult] CheckBroaderGroupInheritanceSettingsForSecureFile ([ControlResult] $controlResult) {
+
+        try {
+            $controlResult.VerificationResult = [VerificationResult]::Failed
+            $projectId = $this.ResourceContext.ResourceDetails.Id
+
+            if ($this.ControlSettings -and [Helpers]::CheckMember($this.ControlSettings, "SecureFile.RestrictedBroaderGroupsForSecureFile") -and [Helpers]::CheckMember($this.ControlSettings, "SecureFile.RestrictedRolesForBroaderGroupsInSecureFile")) {
```
Comment (arvindsingh-cloud) [inline src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.Project.ps1:2655]: not needed first condition if($this.ControlSettings -and

File: src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.Project.ps1 [hunk 4]
```diff
@@ -2645,4 +2645,206 @@ class Project: ADOSVTBase
...
+
+        return $controlResult;
+    }
+
+    hidden [ControlResult] CheckBroaderGroupInheritanceSettingsForRepo ([ControlResult] $controlResult) {        
+        $accessList = @() 
+        $controlResult.VerificationResult = [VerificationResult]::Failed       
+        try{
+
+            if ($this.ControlSettings -and [Helpers]::CheckMember($this.ControlSettings, "SecureFile.RestrictedBroaderGroupsForSecureFile") -and [Helpers]::CheckMember($this.ControlSettings, "SecureFile.RestrictedRolesForBroaderGroupsInSecureFile")) {
```
Comment (arvindsingh-cloud) [inline src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.Project.ps1:2704]: not needed if ($this.ControlSettings -and

**Review by SaikumarRagala — DISMISSED** [review]

**Pull request closed without merging**